server/channels/store/localcachelayer/terms_of_service_layer.go

```diff
@@ -47,11 +47,9 @@ func (s LocalCacheTermsOfServiceStore) Save(termsOfService *model.TermsOfService
 
 func (s LocalCacheTermsOfServiceStore) GetLatest(allowFromCache bool) (*model.TermsOfService, error) {
 	if allowFromCache {
-		if l, err := s.rootStore.termsOfServiceCache.Len(); err == nil && l != 0 {
-			var cacheItem *model.TermsOfService
-			if err := s.rootStore.doStandardReadCache(s.rootStore.termsOfServiceCache, LatestKey, &cacheItem); err == nil {
-				return cacheItem, nil
-			}
+		var cacheItem *model.TermsOfService
+		if err := s.rootStore.doStandardReadCache(s.rootStore.termsOfServiceCache, LatestKey, &cacheItem); err == nil {
```

Comment:
Unrelated but can I ask what's the point of returning an error if we default to using it as a boolean?

Reply:
It's unidiomatic. We need to pass a logger and log the error. I'll need to do that because otherwise one will have no idea if Redis is down.